### ISSUE TITLE
intellisense incorrectly reports no `std` member `queue`

### ISSUE THREAD

**Issue by doug-hoskisson-zwift** [microsoft/vscode-cpptools#9325]
Issue Type: <b>Bug</b>

My code compiles and runs fine, but intellisense reports `namespace "std" has no member "queue" C/C++(135)`

It's ok with other `std` members, like `vector` and `string`.

I was unable to reproduce this with a smaller project.

Extension version: 1.10.2
VS Code version: Code 1.67.1 (da15b6fd3ef856477bf6f4fb29ba1b7af717770d, 2022-05-06T12:37:03.389Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7300U CPU @ 2.60GHz (4 x 2712)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.88GB (0.91GB free)|
|Process Argv|--crash-reporter-id 755a0b02-758d-4f38-9d91-115dd5c22785|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30443607
pythontb:30283811
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
pythonvsnew555:30457759
vscscmwlcmt:30465135
cppdebug:30475216
vsclangdf:30486550

```

</details>

**Comment by doug-hoskisson-zwift** [comment]
![cpp-extension](https://user-images.githubusercontent.com/103061653/169107350-f8ba74b9-3c02-4d35-96a3-d73c4ceedc88.png)

**Comment by sean-mcmanus** [comment]
We need more info, such as from running the C/C++: Log Diagnostics command from the Command Palette or the compiler used, C/C++ std version, etc. The result of doing a Go to Def on the queue could help too.

**Comment by doug-hoskisson-zwift** [comment]
"Go to Definition" moves the cursor to the beginning of that line.
"Go to Declaration" opens the VS build tools header file.
 - Returning to this file after opening the VS build tools header file removes the squiggles.
    - Any edit to the file brings the squiggles back.

<details>
<summary>C/C++: Log Diagnostics</summary>
-------- Diagnostics - 5/18/2022, 10:49:11 AM
Version: 1.10.2
Current Configuration:
{
    "name": "Win32",
    "includePath": [
        "${workspaceFolder}/demos",
        "${workspaceFolder}/source/api",
        "${workspaceFolder}/demos/windows/libs/glfw/include"
    ],
    "defines": [
        "_DEBUG",
        "UNICODE",
        "_UNICODE"
    ],
    "windowsSdkVersion": "10.0.19041.0",
    "compilerPath": "C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe",
    "cStandard": "c17",
    "cppStandard": "c++17",
    "intelliSenseMode": "windows-msvc-x64",
    "compilerPathIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "intelliSenseModeIsExplicit": true,
    "mergeConfigurations": false,
    "browse": {
        "path": [
            "${workspaceFolder}/demos",
            "${workspaceFolder}/source/api",
            "${workspaceFolder}/demos/windows/libs/glfw/include",
            "${workspaceFolder}"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Translation Unit Mappings:
[redacted]
Translation Unit Configurations:
[ D:\[redacted] ]:
    Process ID: 8144
    Memory Usage: 265 MB
    Compiler Path: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe
    Includes:
        D:\[redacted]\DEMOS
        D:\[redacted]\SOURCE\API
        D:\[redacted]\DEMOS\WINDOWS\LIBS\GLFW\INCLUDE
        C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\BUILDTOOLS\VC\TOOLS\MSVC\14.29.30133\INCLUDE
        C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\BUILDTOOLS\VC\TOOLS\MSVC\14.29.30133\ATLMFC\INCLUDE
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\UM
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\UCRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\SHARED
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\WINRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\CPPWINRT
    Defines:
        _DEBUG
        UNICODE
        _UNICODE
    Standard Version: ms_c++17
    IntelliSense Mode: windows-msvc-x64
Total Memory Usage: 531 MB

------- Workspace parsing diagnostics -------
Number of files discovered (not excluded): 5633
Number of files parsed: 1270

</details>

**Comment by sean-mcmanus** [comment]
That logging seems normal.

Does a test.cpp file with just 
```
#pragma once

#include <string>
#include <iostream>
#include <algorithm>
#include <vector>
#include <queue>

std::queue<int> q;
```
work correctly?

Does setting C_Cpp.intelliSenseCacheSize to 0 fix the issue?

Does the issue repro with 1.9.8?

**Comment by doug-hoskisson-zwift** [comment]
The small test file works correctly.

C_Cpp.intelliSenseCacheSize to 0 does not fix the issue.

1.9.8 has the same problem.

**Comment by sean-mcmanus** [comment]
It seems like some code after the std::queue usage is causing our IntelliSense parser to malfunction (i.e. the std::queue definition to be removed), possibly something related to the `#include` statements that come afterwards. We have a bunch of pending IntelliSense fixes for our next 1.10.3 release, but I don't think this is one of them. Are you able to create a minimal repro file you can share? If you run a command like `cl.exe /EP /std:c++17 <any other required flags> source.cpp` that will generate a preprocessed file, which may or may not repro the bug still.
